### PR TITLE
Issue #832 transport unstructured partitioning

### LINKED ISSUE #832: add tests for transport partitioning on unstructured grids
there should be a test that partitions a simulation with flow and 1 or more transport models, on an unstructured grid.
It should feature inactive cells as well.

### PR DESCRIPTION
**Pull request by luitjansl**
Fixes #832 

# Description
Adds a test for partitioning a simulation containing both a flow and  a transport model, including a buoyancy package, 
Includes a fix for partitioning the buoyancy package.
Includes fixes for new mypy checks.

# Checklist
- [X] Links to correct issue
- [ ] Update changelog, if changes affect users
- [X] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [X] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/typing/grid.py [hunk 1]
```diff
@@ -268,3 +268,33 @@ def is_spatial_2D(array: xu.UgridDataArray) -> bool:
...
+    if not array1.shape == array2.shape:
+        return False    
+    if not list(array1.coords.keys())  == list(array2.coords.keys()) :
+        return False
+    for k in list(array1.coords.keys()):
+        if not np.all(array1[k].values == array2[k].values):
+            return False
+    if not np.all(array1.values == array2.values):
+        return False            
+    return True
```
Comment (JoerivanEngelen) [inline imod/typing/grid.py:296]: Why didn't you use the xarray ``DataArray.equals`` method? I think this does what you want and it reduces our codebase. If this is not strict enough, you can use the ``DataArray.identical`` function. See: https://docs.xarray.dev/en/stable/generated/xarray.DataArray.equals.html

File: imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py [hunk 2]
```diff
@@ -405,3 +405,23 @@ def test_partitioning_unstructured_with_well(
     # np.testing.assert_allclose(
     #     cbc["chd"].values, original_cbc["chd"].values, rtol=1e-5, atol=1e-3
     # )
+
+
+@pytest.mark.usefixtures("circle_model_transport")
+@parametrize_with_cases("partition_array", cases=PartitionArrayCases)
+def test_partition_transport(    tmp_path: Path,
+    circle_model_transport: Modflow6Simulation,partition_array: xu.UgridDataArray):
```
Comment (JoerivanEngelen) [inline imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py:413]: Formatting looks off here, did ruff produce this?

File: imod/typing/grid.py [hunk 3]
```diff
@@ -268,3 +268,33 @@ def is_spatial_2D(array: xu.UgridDataArray) -> bool:
...
+def is_equal(array1: xu.UgridDataArray, array2: xu.UgridDataArray)->bool:
+    # TODO: use .equals function of xugrid once new release has been made
+    if not array1.shape == array2.shape:
+        return False
+    if not (array1.values == array2.values).all():
+        return False
+
+    grid_dataset_1 = array1.ugrid.grid.to_dataset()
+    grid_dataset_2 = array2.ugrid.grid.to_dataset()       
+    return grid_dataset_1.equals(grid_dataset_2)
```
Comment (JoerivanEngelen) [inline imod/typing/grid.py:283]: This function is fine for now, but we can remove these functions when the new xugrid gets released. See also my other comment.

**Review by JoerivanEngelen — CHANGES_REQUESTED** [review]
Looks good, good that you are adding extra test cases. I have a few comments though.

File: imod/typing/grid.py (hunk 1 above)
Comment (luitjansl) [reply imod/typing/grid.py:296]: good point, done

File: imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py (hunk 2 above)
Comment (luitjansl) [reply imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py:413]: yes

File: imod/typing/grid.py (hunk 3 above)
Comment (luitjansl) [reply imod/typing/grid.py:283]: indeed, note that there is a #TODO placed at the beginning of the function mentioning that.

**Review by JoerivanEngelen — APPROVED** [review]

**Pull request merged**